c/src/ml-api-inference-single.c

```diff
@@ -1745,8 +1749,33 @@ ml_single_set_property (ml_single_h single, const char *name, const char *value)
       num = gst_tensors_info_parse_types_string (&gst_info, value);
     else if (g_str_has_suffix (name, "name"))
       num = gst_tensors_info_parse_names_string (&gst_info, value);
-    else
-      num = gst_tensors_info_parse_dimensions_string (&gst_info, value);
+    else {
```

Comment:
It looks good.
It seems to be one purpose to bring a dimension to the rank[] list according to input or output.
It would be nice to make it a function as well.
static
gint
xxx(GstTensorsInfo * info, const gchar * dim_string, guint *rank)
{

 return num_dim;
}